### PR TITLE
feat(xds): do not generate independent listener for vips, use additional_addresses instead

### PR DESCRIPTION
**Pull request by jijiechen**
fixes https://github.com/kumahq/kuma/issues/6717

I haven't run a full CI on local env yet, so creating this PR as a draft. 

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues 
  - https://github.com/kumahq/kuma/issues/6717
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS 
  - Confirmed
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) 
  - Done
  - Don't forget `ci/` labels to run additional/fewer tests
  - No need to add
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? 
  - No
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) 
  - No

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/xds/generator/outbound_proxy_generator.go [hunk 1]
```diff
@@ -46,7 +46,13 @@ func (g OutboundProxyGenerator) Generate(ctx context.Context, _ *model.ResourceS
 	// If we have same split in many HTTP matches we can use the same cluster with different weight
 	clusterCache := map[string]string{}
 
+	serviceVips, serviceToVipsMap := buildServiceToVipsMap(outbounds, xdsCtx.Mesh.VIPOutbounds)
```
Comment (lukidzi) [inline pkg/xds/generator/outbound_proxy_generator.go:49]: @michaelbeaumont Do we need the same logic in Mesh*Route?

File: pkg/xds/envoy/listeners/listener_configurers.go [hunk 2]
```diff
@@ -97,3 +97,13 @@ func TagsMetadata(tags map[string]string) ListenerBuilderOpt {
 		Tags: tags,
 	})
 }
+
+func AdditionalAddresses(addresses []mesh_proto.OutboundInterface) ListenerBuilderOpt {
+	if len(addresses) < 1 {
+		return ListenerBuilderOptFunc(nil)
+	}
+
+	return AddListenerConfigurer(&v3.AdditionalAddressConfigurer{
```
Comment (michaelbeaumont) [inline pkg/xds/envoy/listeners/listener_configurers.go:106]: already handled

```suggestion
	return AddListenerConfigurer(&v3.AdditionalAddressConfigurer{
```

File: pkg/xds/generator/outbound_proxy_generator.go [hunk 3]
```diff
@@ -441,3 +448,37 @@ func (OutboundProxyGenerator) determineRoutes(
...
 }
+
+func buildServiceToVipsMap(outbounds []*mesh_proto.Dataplane_Networking_Outbound, meshVIPOutbounds []*mesh_proto.Dataplane_Networking_Outbound) (map[string]bool, map[string][]mesh_proto.OutboundInterface) {
+	vipMap := map[string][]mesh_proto.OutboundInterface{}
+	var dpNetworking *mesh_proto.Dataplane_Networking
+
+	serviceVips := map[string]bool{}
+	for _, ob := range meshVIPOutbounds {
+		service := ob.GetService()
+		if service != "" {
```
Comment (michaelbeaumont) [inline pkg/xds/generator/outbound_proxy_generator.go]: IMO checks like this are just noise over the algorithm, we can assume a non empty service here.

**Review by michaelbeaumont — APPROVED** [review]

**Comment by jijiechen** [comment]
I'm working on the E2E failures.

File: pkg/xds/generator/outbound_proxy_generator.go (hunk 3 above)
Comment (jijiechen) [reply pkg/xds/generator/outbound_proxy_generator.go]: I've seen this in somewhere else, so I just added.

**Pull request merged**